### ISSUE TITLE
Dynamic icons are gone

### ISSUE THREAD

**Issue by Eagle3386** [notepad-plus-plus/nppShell#52]
Back in the days of `NppShell_06.dll`, running `regsvr32 /i NppShell_06.dll` opened up a dialog for, among other things, enabling the "dynamic icon" feature:
![image](https://github.com/notepad-plus-plus/nppShell/assets/5111490/b9ef0031-ac46-4683-8f08-1afe336f50b8)

This would turn any associated file's icon into something like this:
![202388990-f408d3f8-03fe-46ae-b75f-571d18ddefdb](https://github.com/notepad-plus-plus/nppShell/assets/5111490/9c717cc1-d236-442c-bf14-de6a7ef3409e)
(Image borrowed from notepad-plus-plus/notepad-plus-plus/issues/12531)

Was this _awesome_ feature dropped from the new `NppShell.dll`?
If yes, can you add it back, pretty please? 🥺
If not & it's still there, how to enable it?

**Comment by GurliGebis** [comment]
Since the new implementation is using the IExplorerCommand interface, it is not something we really can do.

**Comment by Eagle3386** [comment]
Sorry for my lack of knowledge, but I'm curious: why's the new interface blocking re-adding that feature?

Also, given that limitation, couldn't N++ just _additionally_ support the "old" interface that's used for those dynamic icons, as well?

**Comment by GurliGebis** [comment]
The old implementation used the old `IContextMenu3` interface, while the new `IExplorerCommand` interface.
So adding it isn't something that easily can be done, since they work completely differently.

**Comment by Eagle3386** [comment]
@GurliGebis / @donho
Sorry to resurrect, but couldn't make it sooner here.
Regarding the topic: couldn't we combine both approaches, using `IExplorerCommand` for the context menu part & `IContextMenu3` for the dynamic icon related stuff, then?

Also, as I'm no C++ expert, but still curious to understand the internals, what exactly is missing in the new interface that the old offered to enable the dynamic icon handling?
Because reading through MS docs regarding both interfaces, I don't see _anything_ that could be used for dynamic icons - yet, it was available a couple of months ago..?! 🤔

**Comment by GurliGebis** [comment]
@Eagle3386 not really from how I see it.
Instead what I would suggest people do is enable file extensions in explorer (which should be the default in this day and age).

**Comment by Eagle3386** [comment]
I've got that enabled since I started using Windows (back in the days of 3.11 & 95 SE, that is), but I still find it _way_ more accessible, easier, more comfortable & even more aesthetical to have icons represent what they files actually contain.

IMHO the alternative, or should I say fix, would be to include a specific icon for each supported filetype - which could contain the file extension as it previously did.

Another alternative would be to include the software's logo as an overlay on top of N++'s "page" icon.
